native/dlasq2.go

```diff
@@ -0,0 +1,326 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package native
+
+import (
+	"math"
+
+	"github.com/gonum/lapack"
+)
+
+// Dlasq2 computes all the eigenvalues of the symmetric positive
+// definite tridiagonal matrix associated with the qd array Z to high
+// relative accuracy are computed to high relative accuracy, in the
+// absence of denormalization, underflow and overflow.
+//
+// To see the relation of Z to the tridiagonal matrix, let L be a
+// unit lower bidiagonal matrix with subdiagonals Z(2,4,6,,..) and
+// let U be an upper bidiagonal matrix with 1's above and diagonal
+// Z(1,3,5,,..). The tridiagonal is L*U or, if you prefer, the
+// symmetric tridiagonal to which it is similar.
+//
+// info returns a status error.
+func (impl Implementation) Dlasq2(n int, z []float64) (info int) {
+
+	if len(z) < 4*n {
+		panic("lapack: bad z")
```

Comment:
I know that this is phrased similarly to "bad work", but it's very similar to the message below. Can one or both be changed to make a clearer distinction. Maybe change those below to "negative z value" (use a constant).